### PR TITLE
can_exec(): move from misc_macros to misc_patterns

### PR DESCRIPTION
**Pull request by cgzones**
The file misc_macros.spt is due heavy usage of the m4 language
hard to parse for third party tools.
Move the macro can_exec() to misc_patterns.spt, which contains
only interface like define blocks.

### CODE REVIEW

**Comment by doverride** [comment]
cgzones <notifications@github.com> writes:

> The file misc_macros.spt is due heavy usage of the m4 language
> hard to parse for third party tools.
> Move the macro can_exec() to misc_patterns.spt, which contains
> only interface like define blocks.

the existing exec_files_pattern() is can_exec()'s pattern equivalent

>
> --------------------------------------------------------------------------------------------------------------------------------------
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/SELinuxProject/refpolicy/pull/260
>
> Commit Summary
>
> * can_exec(): move from misc_macros to misc_patterns
>
> File Changes
>
> * M policy/support/misc_macros.spt (6) 
> * M policy/support/misc_patterns.spt (9) 
>
> Patch Links:
>
> * https://github.com/SELinuxProject/refpolicy/pull/260.patch
> * https://github.com/SELinuxProject/refpolicy/pull/260.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

-- 
gpg --locate-keys dominick.grift@defensec.nl
Key fingerprint = FCD2 3660 5D6B 9D27 7FC6  E0FF DA7E 521F 10F6 4098
https://sks-keyservers.net/pks/lookup?op=get&search=0xDA7E521F10F64098
Dominick Grift

**Comment by cgzones** [comment]
`exec_files_pattern()` is not exactly an equivalent, cause it requires the type of the containing directory, which in many cases is `bin_t` and therefore not in scope.

**Comment by dburgener** [comment]
Every single pattern in the support directory consists of multiple lines, and ends with the suffix "_pattern".  It seems to me as though can_exec() fits more at home in the macros (although it is a bit of a corner case in there as well since it I believe is the only one that generates an allow rule).

I'd say that third party tools can probably handle this situation fairly easily by just treating can_exec() as a special case.

**Comment by doverride** [comment]
cgzones <notifications@github.com> writes:

> exec_files_pattern() is not exactly an equivalent, cause it requires the type of the containing directory, which in many cases is
> bin_t and therefore not in scope.

True, but that is what makes it a pattern and why can_exec() does not
belong there

>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

-- 
gpg --locate-keys dominick.grift@defensec.nl
Key fingerprint = FCD2 3660 5D6B 9D27 7FC6  E0FF DA7E 521F 10F6 4098
https://sks-keyservers.net/pks/lookup?op=get&search=0xDA7E521F10F64098
Dominick Grift
